### DISCUSSION TITLE
`form` only populating after second submission

### DISCUSSION THREAD

**Discussion by Creative-Difficulty** [sveltejs/kit#12383] [Q&A]
### Describe the bug

The exact line: https://github.com/Creative-Difficulty/marc-plan/blob/8c0ca475d086740cae206aa608d58022429871d5/src/routes/signup/%2Bpage.svelte#L122
After the first form submission, the actions runs as normal, but the export let form: ActionData doesn't populate and stays (or becomes?) null. You can check this, by logging the form variable after submit and before the await update(...) function is called, as marked in the link above. This means, that errors get returned from the action (this can be confirmed by replacing the console.log in the code with console.log(result) and seeing the data.message equals the message returned from the server), but the form variable doesn't get populated. I think this only affects the signup route, but i haven't checked yet.

Another question: If i call the toast.error(...) after the await update(...), it doesn't run, but if i call it before, everything works as normal.
My theory: update(...) populates form , so when logging it before calling update(...), it still has the "old" state (from the previous submission) which in case of the first submission would be null. Is this true? Is this documented anywhere? Any help is appreciated!

I might also be using progressive enhancement wrong in general.

### Reproduction

GitHub repo: https://github.com/Creative-Difficulty/marc-plan/
A turso database has to be created for full reproduction.

### System Info

```Shell
System:
    OS: macOS 14.4.1
    CPU: (10) arm64 Apple M2 Pro
    Memory: 600.42 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.7.0 - ~/.nvm/versions/node/v20.7.0/bin/node
    npm: 10.2.5 - ~/.nvm/versions/node/v20.7.0/bin/npm
    pnpm: 9.1.1 - ~/Library/pnpm/pnpm
    bun: 1.1.12 - ~/.bun/bin/bun
    Watchman: 2024.05.06.00 - /opt/homebrew/bin/watchman
  Browsers:
    Safari: 17.4.1
  npmPackages:
    @sveltejs/adapter-auto: ^3.0.0 => 3.2.1 
    @sveltejs/kit: ^2.0.0 => 2.5.10 
    @sveltejs/vite-plugin-svelte: ^3.0.0 => 3.1.1 
    svelte: ^4.2.7 => 4.2.18 
    vite: ^5.0.3 => 5.2.13
```

### Severity

serious, but I can work around it

**Comment by teemingc** [comment]
You need to call `update` or `applyAction` first to update the form prop.

> The behaviour of applyAction(result) depends on result.type:
> 
> success, failure — sets $page.status to result.status and updates form and $page.form to result.data (regardless of where you are submitting from, in contrast to update from enhance)

https://kit.svelte.dev/docs/form-actions#progressive-enhancement-customising-use-enhance
